app/js/docker/dockerService.js

```diff
@@ -66,6 +66,11 @@ function dockerService($http, actions, flux, alertService, configService) {
             }
         };
 
+        if (action === 'pullImage') {
+            // hack for pulling images
+            config.params = data;
+        }
+
```

Comment:
The other option is to restructure how we call into this service. One potential API could be:

```
dockerService.do({
    url: '/images/pull',
    host: 'hostname',
    query: { query param map },   // both query and data keys are optional
    data: ...                                   // and more importantly, not mutually exclusive :)
});
```

This would require the dockerService itself to maintain a map from API urls to actions and methods (which we kind of do already anyway)

Which option sounds more appealing?